dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/injection/EEValidationUtils.java

```diff
@@ -166,7 +166,14 @@ private static void validateJndiLookup(String lookupString, Annotated annotated,
         // We need to set a current component before doing anything to do with JNDI
         CDIRuntime cdiRuntime = cdiArchive.getCDIRuntime();
 
-        try (ContextBeginnerEnder cbe = cdiRuntime.createContextBeginnerEnder().extractComponentMetaData(cdiArchive).beginContext()) {
+        //This method can be invoked as part of application startup, if so don't set a context when one is already active.
+        ContextBeginnerEnder cbe = null;
+
+        try {
+            if (! cdiRuntime.isContextBeginnerEnderActive()) {
+                cbe = cdiRuntime.createContextBeginnerEnder().extractComponentMetaData(cdiArchive).beginContext(); //TODO make this into a  when we start compiling with java 9
```

Comment:
Does anything need doing with this comment? and what is this supposed to be made into, as that is missing.

Reply:
Good catch. Its try-with-resources. See: https://stackoverflow.com/questions/28702219/how-to-use-try-with-resources-with-if-statement

I thought the java8 workaround was too ugly